temporalcli/commands.workflow.go

```diff
@@ -96,16 +128,26 @@ func (*TemporalWorkflowUpdateCommand) run(*CommandContext, []string) error {
 	return fmt.Errorf("TODO")
 }
 
+type singleOrBatchParams struct {
+	AllowReasonWithWorkflowID bool
+}
+
 func (s *SingleWorkflowOrBatchOptions) workflowExecOrBatch(
 	cctx *CommandContext,
 	namespace string,
 	cl client.Client,
+	options ...func(*singleOrBatchParams),
```

Comment:
Can you use an options struct instead of this mutator pattern? We typically don't use this pattern in our codebase.

Reply:
We use it often in the server; good to know the SDK/CLI code doesn't. I'd hoped to use this so that we didn't need to change callsites which relied on the default behavior

I'll make that change though

Reply:
I wouldn't worry the other callsites in internal code.
Thanks for changing.